### ISSUE TITLE
EventGridTrigger batching not working

### ISSUE THREAD

**Issue by nixa333** [Azure/azure-functions-dotnet-worker#736]
I'm currently unable to get EventGridTrigger to work with batches.
My function uses .NET 5.0 dotnet-isolated runtime with the following trigger:
```csharp
public async Task Run([EventGridTrigger] EventGridTriggerData eventGridEvent, FunctionContext context)
```

I've tried using 
```csharp
[EventGridTrigger] EventGridTriggerData[] eventGridEvents
```
but that just crashes the function.

`EventGridTriggerData` is
```csharp
public class EventGridTriggerData
{
    public string Id { get; set; }
    public string Topic { get; set; }
    public string Subject { get; set; }
    public string EventType { get; set; }
    public DateTime EventTime { get; set; }
    public IDictionary<string, object> Data { get; set; }
}
```

I've set up the Event subscription on my Azure BLOB storage account and enabled batching, however my code is triggered once per each BLOB that lands in the container I'm subscribed to (Subscription is only for "Blob Created" event and created with **EventGridSchema** schema).
![image](https://user-images.githubusercontent.com/6515226/144013151-b8dc2811-affa-424a-93aa-65085322db7b.png)

Any advice here?

**Comment by fabiocav** [comment]
@nixa333 we'll need to investigate this before we can provide any guidance, but it does seem to be a defect if this is happening as described. We'll work on setting up a repro, but if you have one built, that would help ensure we're validating the same flow. 

Can you also share the version of the Event Grid package you're using?

**Comment by nixa333** [comment]
@fabiocav the code is actually as simple as what I've attached in the original post. Plain .NET 5.0 function with `EvebtGridTrigger` does not work.

```csharp
[Function("Function")]
public async Task Run([EventGridTrigger] EventGridTriggerData eventGridEvent, FunctionContext context)
```

Here's a screenshot of all nugets with versions
![image](https://user-images.githubusercontent.com/6515226/144246259-6e8c9d7c-de07-459e-bbaa-80542ae8a871.png)

And important part of the `.csproj`
```XML
<PropertyGroup>
	<TargetFramework>net5.0</TargetFramework>
	<AzureFunctionsVersion>v3</AzureFunctionsVersion>
	<OutputType>Exe</OutputType>
</PropertyGroup>
```

host.json
```JSON
{
    "version": "2.0",
    "logging": {
        "applicationInsights": {
            "samplingSettings": {
                "isEnabled": true,
                "excludedTypes": "Request"
            }
        }
    },
    "functionTimeout": "01:00:00"
}
```

As far as the Azure part goes, I have a Locally redundant StorageV2 (general purpose v2),
Event Grid System Topic:
![image](https://user-images.githubusercontent.com/6515226/144247480-598b31e1-12c5-44d2-ac65-820c1be35e18.png)
And an event subscription created directly through storage account with filtering set to react only to a specific container:
![image](https://user-images.githubusercontent.com/6515226/144247894-f5846ffc-2c60-4e29-aadb-0bcdc0d89c3a.png)


with **Web Hook** endpoint **https://....azurewebsites.net/runtime/webhooks/EventGrid?functionName=Function** 

![image](https://user-images.githubusercontent.com/6515226/144248675-50844bc1-73e7-4c79-9be4-dc511bf99c17.png)
and Additional features set as seen in the original post screenshot

Azure function config:
![image](https://user-images.githubusercontent.com/6515226/144249614-bfc11f2c-46cd-4e31-b4df-487c207442b4.png)

**Comment by fabiocav** [comment]
Adding this to the backlog for sprint assignment.

**Comment by jblanchet** [comment]
Any idea when this will be addressed? Or is there another way to implement batching with event grid triggered functions that works around this?

**Comment by gabrielrainha** [comment]
@nixa333 @jblanchet 
I was experiencing the same problem. but on .NET 6.0.
Today I managed to fix it by upgrading to .NET 7.0, upgrading the Extensions.EventGrid to the latest (3.3.0) and setting a flag on the EventGridTriggerAttribute, like this:

`public async Task Run([EventGridTrigger(IsBatched = true)] EventGridEvent[] eventGridEvents, FunctionContext context)`

Such flag was introduced on version 3.3.0, Not sure upgrading do .NET 7.0 was required, but I did it nonetheless.

Here's a screenshot of the nuggets I used:
![image](https://github.com/Azure/azure-functions-dotnet-worker/assets/11670858/d76eb30a-0afc-445f-a351-02404a39d178)

[Here's the sample code](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/samples/WorkerBindingSamples/EventGrid/EventGridEventSamples.cs) that led me to this solution:
